datafusion/core/src/datasource/file_format/parquet.rs

```diff
@@ -818,14 +826,16 @@ impl DataSink for ParquetSink {
 async fn column_serializer_task(
     mut rx: Receiver<ArrowLeafColumn>,
     mut writer: ArrowColumnWriter,
-) -> Result<ArrowColumnWriter> {
+    mut reservation: MemoryReservation,
+) -> Result<(ArrowColumnWriter, MemoryReservation)> {
     while let Some(col) = rx.recv().await {
         writer.write(&col)?;
+        reservation.try_resize(writer.memory_size())?;
     }
-    Ok(writer)
+    Ok((writer, reservation))
```

Comment:
Steps for tracking of parallelized write usage.

In function above (`column_serializer_task`):
    1. track when add to ArrowColumnWriters
    2. yield reservation (so doesn't drop & free)

In `spawn_rg_join_and_finalize_task`:
    3. this performs ArrowColumnWriters.close() and yields the encoded ColumnChunk. Therefore resize the col reservation, to the encoded size.

In `concatenate_parallel_row_groups`
    4. this serializes the row groups. Therefore add the encoded size to the rg_reservation, and drop each col_reservation.

Reply:
I feel like that^^ should be documented somewhere. On which struct?